File: Frontend/src/converters/ImportantItemsConverter.ts
import type * as DTO from '../events/dto/events.dto';
import type { ImportantItems } from '../models/Items';

export class ImportantItemsConverter {
    public static convert(importantItems: DTO.ImportantItemsDTO | null): ImportantItems | null {
        if (!importantItems) return null;
        return { ...importantItems };
    }
}

File: Backend/Events/Converters/ImportantItemsConverter.cs
using Backend.Domain.Models;
using Backend.Events.DTO;

namespace Backend.Events.Converters;

public static class ImportantItemsConverter
{
    public static ImportantItemsDTO ToDTO(ImportantItems importantItems)
    {
        return new ImportantItemsDTO
        {
            TreeBoots = importantItems.TreeBoots,
            FishingPole = importantItems.FishingPole,
            AsukaTrophy = importantItems.AsukaTrophy,
            SunTrophy = importantItems.SunTrophy
        };
    }
}
